### PR TITLE
Scheduler queries

### PR DESCRIPTION
**Pull request by plauche**
Implementing the `activeSchedule` and `registeredSchedule` queries and updating tests as appropriate.

### CODE REVIEW

**Comment by plauche** [comment]
[Implement scheduler queries](https://trello.com/c/BO1CYPQ1/1049-implement-scheduler-queries)

File: services/scheduler-service/src/objects.rs [hunk 1]
```diff
@@ -28,3 +33,33 @@ pub struct Schedule {
...
     pub active: bool,
 }
+
+impl Schedule {
+    pub fn from_path(path: &str) -> Result<Schedule, String> {
+        let data =
+            fs::metadata(&path).map_err(|e| format!("Failed to read file metadata: {}", e))?;
+
+        let time_registered: DateTime<Utc> = data
+            .modified()
```
[the remark this answers is not in the document]
Comment (plauche) [reply services/scheduler-service/src/objects.rs]: Weirdly...I get this error `creation time is not available on this platform currently`.

File: services/scheduler-service/src/scheduler.rs [hunk 2]
```diff
@@ -77,4 +78,62 @@ impl Scheduler {
...
+        for path in fs::read_dir(&self.scheduler_dir)
+            .map_err(|e| format!("Failed to read schedules dir: {}", e))?
+            // Filter out invalid entries
+            .filter_map(|x| x.ok())
+            // Convert DirEntry -> PathBuf
+            .map(|entry| entry.path())
+            // Filter out non-files
+            .filter(|entry| entry.is_file())
+            // Filter out active.json
+            .filter(|path| !path.ends_with("active.json"))
```
[the remark this answers is not in the document]
Comment (plauche) [reply services/scheduler-service/src/scheduler.rs]: I found that it was not filtered out :(

[the remark this answers is not in the document]
Comment (CatGarab) [reply services/scheduler-service/src/scheduler.rs]: Oh weird. Carry on then :P

**Pull request merged**